src/Client.sol

```diff
@@ -30,17 +41,93 @@ contract Client is Initializable, IClient, MulticallUpgradeable, Ownable2StepUpg
      * @param params The parameters for the transfer
      * @dev Reverts with InsufficientAllowance if caller doesn't have sufficient allowance
      * @dev Reverts with InvalidAmount when parsing amount from BigInt to uint256 failed
+     * @dev Reverts with UnfairDistribution when trying to give too much to single SP
      */
     function transfer(DataCapTypes.TransferParams calldata params) external {
         (uint256 parsedAmount, bool failed) = BigInts.toUint256(params.amount);
         if (failed) revert Errors.InvalidAmount();
         if (allowances[msg.sender] < parsedAmount) revert Errors.InsufficientAllowance();
-        uint64[] memory providers = _deserializeAllocationRequests(params.operator_data);
+        (uint64[] memory providers, uint256[] memory sizes) = _deserializeAllocationRequests(params.operator_data);
         _ensureSPsAreAllowed(providers);
+
+        uint256 totalSize = 0;
+        for (uint256 i = 0; i < sizes.length; i++) {
+            totalSize += sizes[i];
```

Comment:
@Filip-L , when testing this in practice please check if `totalSize == parsedAmount` for real calls. Will says it does. If yes, let's drop this loop and just use `parsedAmount` as `totalSize`.
Can you create a JIRA ticket for this so that we don't forget?